### PR TITLE
Address a few gcc 6.x warnings

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

File: core/lib/libtomcrypt/src/ciphers/sub.mk [hunk 1]
```diff
@@ -11,7 +11,6 @@ srcs-y += aes_armv8a_ce.c
 srcs-y += aes_modes_armv8a_ce_a32.S
 else
 srcs-$(CFG_CRYPTO_AES) += aes.c
-srcs-$(CFG_CRYPTO_AES) += aes_tab.c
```
Comment (jforissier) [inline core/lib/libtomcrypt/src/ciphers/sub.mk]: Should we also delete the file from Git? I have no strong opinion, but I'm leaning towards deleting it.

Comment (jenswi-linaro) [inline core/lib/libtomcrypt/src/ciphers/sub.mk]: We should keep the file, it's included by `aes.c` above.

**Comment by jenswi-linaro** [comment]
Tag applied

**Comment by jenswi-linaro** [comment]
No need to upstream, the changes is only in code we've written ourselves.

**Pull request merged**